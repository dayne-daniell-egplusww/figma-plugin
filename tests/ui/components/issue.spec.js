import { shallowMount } from '@vue/test-utils';
import Issue from '~/ui/components/issue';
import FigmaIcon from '~/ui/components/figma_plugin_ds/icon';
import mockIssue from '../mock_data/issue';

describe('Issue component', () => {
  let wrapper;

  function createComponent({ issue = mockIssue, selected } = {}) {
    wrapper = shallowMount(Issue, {
      propsData: {
        issue,
        selected,
      },
    });
  }

  it('renders the component', () => {
    createComponent();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits click event when clicked', () => {
    createComponent();
    wrapper.trigger('click');
    expect(wrapper.emitted().click).toBeTruthy();
  });

  describe('with a closed issue', () => {
    const closedIssue = {
      ...mockIssue,
      state: 'closed',
    };

    it('renders indication that the issue is closed', () => {
      createComponent({ issue: closedIssue });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('when selected', () => {
    beforeEach(() => {
      createComponent({ selected: true });
    });
    it('adds background color', () => {
      expect(wrapper.classes()).toContain('bg-selection-b');
    });
    it('renders check icon', () => {
      const icon = wrapper.findComponent(FigmaIcon);
      expect(icon.exists()).toBe(true);
      expect(icon.props('icon')).toBe('check');
    });
  });
});
