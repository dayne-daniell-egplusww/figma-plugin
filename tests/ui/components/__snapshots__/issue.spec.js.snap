// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue component renders the component 1`] = `
"<li id=\\"issue_54321\\" class=\\"pl-large pr-xxsmall pt-xxsmall pb-xxsmall hover-bg-selection-b cursor-pointer list-style-none position-relative\\">
  <!---->
  <div class=\\"d-flex\\">
    <div class=\\"flex-grow-1\\">
      <div><span><a target=\\"_blank\\" href=\\"https://gitlab.com/gitlab-org/gitlab/-/issues/12345\\" class=\\"text-black8 hover-text-black8 text-large font-weight-bold mb-xxxsmall inline-block line-height-small\\">Merge requests requiring attention</a></span></div>
      <div class=\\"font-weight-normal text-black6 text-small\\"><span class=\\"issue-reference-path\\"></span></div>
    </div>
    <div>
      <!---->
    </div>
  </div>
</li>"
`;

exports[`Issue component with a closed issue renders indication that the issue is closed 1`] = `
"<li id=\\"issue_54321\\" class=\\"pl-large pr-xxsmall pt-xxsmall pb-xxsmall hover-bg-selection-b cursor-pointer list-style-none position-relative\\">
  <!---->
  <div class=\\"d-flex\\">
    <div class=\\"flex-grow-1\\">
      <div><span><a target=\\"_blank\\" href=\\"https://gitlab.com/gitlab-org/gitlab/-/issues/12345\\" class=\\"text-black8 hover-text-black8 text-large font-weight-bold mb-xxxsmall inline-block line-height-small\\">Merge requests requiring attention</a></span></div>
      <div class=\\"font-weight-normal text-black6 text-small\\"><span class=\\"issue-reference-path\\"></span></div>
    </div>
    <div><span class=\\"ml-xxsmall text-large line-height-small\\">CLOSED</span></div>
  </div>
</li>"
`;
